### PR TITLE
Replace X icon with button in audience creation notice.

### PR DESCRIPTION
**Pull request by ankitrox**
## Summary

Addresses issue:

- #9980

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.
- [x] Ensure there are no unexpected significant changes to file sizes.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSelectionPanel/AudienceCreationNotice.js [hunk 1]
```diff
@@ -46,8 +46,7 @@ import {
 	SITE_KIT_AUDIENCE_DEFINITIONS,
 } from '@/js/modules/analytics-4/datastore/constants';
 import { ERROR_CODE_MISSING_REQUIRED_SCOPE } from '@/js/util/errors';
-import Link from '@/js/components/Link';
-import CloseIcon from '@/svg/icons/close.svg';
+import Button from '@/js/googlesitekit/components-gm2/Button';
```
Comment (nfmohit) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSelectionPanel/AudienceCreationNotice.js]: Shouldn't this be imported from `googlesitekit-components` instead?


```suggestion
import { Button } from 'googlesitekit-components';
```

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSelectionPanel/AudienceCreationNotice.js [hunk 2]
```diff
@@ -329,6 +320,15 @@ export default function AudienceCreationNotice() {
 							</div>
 						</div>
 					) ) }
+				<div className="googlesitekit-audience-selection-panel__audience-creation-notice-dismiss">
+					<Button
+						className="googlesitekit-audience-selection-panel__audience-creation-notice-no-thanks"
```
Comment (nfmohit) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSelectionPanel/AudienceCreationNotice.js]: It looks like this class name is redundant. Can we remove it?


```suggestion
```

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSelectionPanel/AudienceCreationNotice.test.js [hunk 3]
```diff
@@ -464,4 +465,93 @@ describe( 'AudienceCreationNotice', () => {
...
+
+		const { container, waitForRegistry } = render(
+			<AudienceCreationNotice />,
+			{ registry }
+		);
+
+		await waitForRegistry();
+
+		expect( container ).toBeEmptyDOMElement();
+	} );
```
Comment (nfmohit) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSelectionPanel/AudienceCreationNotice.test.js]: This test case looks redundant. This technically doesn't simulate clicking the "No thanks" action.

Also, this exact same test is already covered above, see the case named `should render null if the user has dismissed the notice`.


```suggestion
```

File: assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-selection-panel.scss [hunk 4]
```diff
@@ -133,7 +133,9 @@
 
 		.googlesitekit-audience-selection-panel__audience-creation-notice-header {
 			display: flex;
-			justify-content: space-between;
+
+			// No close icon now; keep layout start aligned.
+			justify-content: flex-start;
 		}
```
Comment (nfmohit) [inline assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-selection-panel.scss]: Is this necessary anymore? Can't we remove the styling for `.googlesitekit-audience-selection-panel__audience-creation-notice-header` altogether as it now has only one child?


```suggestion
```

File: assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-selection-panel.scss [hunk 5]
```diff
@@ -146,13 +148,7 @@
...
 
-		.googlesitekit-audience-selection-panel__audience-creation-notice-close {
-			svg {
-				color: $c-surfaces-on-surface-variant;
-				height: 12px;
-				width: 12px;
-			}
-		}
+		// Removed close (X) button styles as per enhancement/9980.
```
Comment (nfmohit) [inline assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-selection-panel.scss]: This comment looks redundant.


```suggestion
```

File: assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-selection-panel.scss [hunk 6]
```diff
@@ -208,6 +204,23 @@
...
+
+			.mdc-button {
+				color: $c-surfaces-on-surface;
+				min-height: auto;
+				padding-right: 0;
+
+				&:hover {
+					background-color: transparent;
+					color: $c-surfaces-on-surface-variant;
+				}
```
Comment (nfmohit) [inline assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-selection-panel.scss]: Since we're adapting the button's style to resemble a link, why not use the `Link` component instead of the `Button`? I feel that would better suit this purpose.

Also, I'm struggling to determine what the spacing between the link and the bottom padding is in the Figma designs (see screenshot), so that is probably worth checking with Sigal.

Image

**Review by nfmohit — CHANGES_REQUESTED** [review]
Nice work on this, thank you @ankitrox!

I've left a few comments for your consideration. Please let me know if you have any questions. Thanks!

File: assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-selection-panel.scss (hunk 6 above)
Comment (ankitrox) [reply assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-selection-panel.scss]: Thanks @nfmohit.

Asked Sigal about the same [here](https://www.figma.com/design/7pSrkEy8t00BcYRAi9LjjH?node-id=6631-18147&m=dev#1485014581), but as of now, I have added 14px vertical padding to the link. Also, used `Link` component instead of `Button` as suggested.

**Comment by ankitrox** [comment]
Thanks @nfmohit for reviewing the PR. I've updated the PR as per suggestion. Over to you for another pass.

File: assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-selection-panel.scss [hunk 7]
```diff
@@ -208,6 +195,26 @@
 		.googlesitekit-audience-creation-error-notice {
 			margin-bottom: 12px;
 		}
+
+		.googlesitekit-audience-selection-panel__audience-creation-notice-dismiss {
+			display: flex;
+			justify-content: flex-end;
```
Comment (nfmohit) [inline assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-selection-panel.scss]: Can't we simply use `text-align` here instead of creating a flex container?


```suggestion
			text-align: end;
```

File: assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-selection-panel.scss [hunk 8]
```diff
@@ -208,6 +195,26 @@
...
+			margin-top: 4px;
```
Comment (nfmohit) [inline assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-selection-panel.scss]: According to the Figma designs, the spacing between the link and the audience above is 14px. Is there a reason we're setting this to 4px?

Image


```suggestion
			margin-top: 14px;
```

File: assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-selection-panel.scss [hunk 9]
```diff
@@ -208,6 +195,26 @@
...
+
+			.googlesitekit-cta-link {
+				color: $c-surfaces-on-surface;
+				font-weight: 500;
+
+				&:hover {
+					text-decoration: none;
+				}
```
Comment (nfmohit) [inline assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-selection-panel.scss]: IMO, we should keep the hover indicator for accessibility.




```suggestion
```

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSelectionPanel/AudienceCreationNotice.js [hunk 10]
```diff
@@ -329,6 +320,11 @@ export default function AudienceCreationNotice() {
 							</div>
 						</div>
 					) ) }
+				<div className="googlesitekit-audience-selection-panel__audience-creation-notice-dismiss">
+					<Link onClick={ onCloseClick } linkButton>
+						{ __( 'No thanks', 'google-site-kit' ) }
+					</Link>
```
Comment (nfmohit) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSelectionPanel/AudienceCreationNotice.js]: According to the Figma designs, the link should hug the container. See screenshot:

Image

The `linkButton` prop adds a padding which defies this requirement. See screenshot:

Image


```suggestion
					<Link onClick={ onCloseClick }>
						{ __( 'No thanks', 'google-site-kit' ) }
					</Link>
```

File: assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-selection-panel.scss [hunk 11]
```diff
@@ -208,6 +195,26 @@
...
+
+				&:active,
+				&:focus {
+					background-color: transparent;
+				}
```
Comment (nfmohit) [inline assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-selection-panel.scss]: If we remove the `linkButton` prop, we may also be able to remove these styles.


```suggestion
```

File: tests/backstop/reference/google-site-kit_Modules_Analytics4_Components_AudienceSegmentation_Dashboard_AudienceSelectionPanel_Audience_creation_notice_with_OAuth_error_0_document_2_large.png
Comment (nfmohit) [inline tests/backstop/reference/google-site-kit_Modules_Analytics4_Components_AudienceSegmentation_Dashboard_AudienceSelectionPanel_Audience_creation_notice_with_OAuth_error_0_document_2_large.png:1]: We should address the spacing between the dismiss link and the error message here. This could follow the same spacing as the warning notice in the same component.

Image

**Comment by ankitrox** [comment]
Thanks for reviewing the PR @nfmohit 

I've made the suggested changes and sending your way for another pass.

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSelectionPanel/AudienceCreationNotice.js (hunk 10 above)
Comment (ankitrox) [reply assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSelectionPanel/AudienceCreationNotice.js]: Removed `linkButton` prop.

File: assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-selection-panel.scss (hunk 11 above)
Comment (ankitrox) [reply assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-selection-panel.scss]: Removed these styles.

File: assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-selection-panel.scss (hunk 9 above)
Comment (ankitrox) [reply assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-selection-panel.scss]: Removed.

File: assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-selection-panel.scss (hunk 8 above)
Comment (ankitrox) [reply assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-selection-panel.scss]: Thank you, updated it.

File: assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-selection-panel.scss (hunk 7 above)
Comment (ankitrox) [reply assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-audience-selection-panel.scss]: That's a nice suggestion. Thanks, updated it 🙂

File: tests/backstop/reference/google-site-kit_Modules_Analytics4_Components_AudienceSegmentation_Dashboard_AudienceSelectionPanel_Audience_creation_notice_with_OAuth_error_0_document_2_large.png
Comment (nfmohit) [inline tests/backstop/reference/google-site-kit_Modules_Analytics4_Components_AudienceSegmentation_Dashboard_AudienceSelectionPanel_Audience_creation_notice_with_OAuth_error_0_document_2_large.png:1]: Thank you for addressing the spacing issue here.

Upon closer inspection, the current implementation leaves an odd amount of spacing at the bottom of the error/warning notice, as the component now has increased bottom padding.

According to the Figma designs, the space between the error/warning notice and the component's bottom should be 18px. Since we already have a 20px padding, let's remove the bottom spacing.

**Review by nfmohit — CHANGES_REQUESTED** [review]
Thank you for addressing my feedback, @ankitrox!

I've left one last comment, but to avoid the round-trip, I'll go ahead and make this change myself.

Thanks!

**Review by nfmohit — APPROVED** [review]
LGTM ✅ 

I've verified that the CI test failures are unrelated to the changes in this PR.
